python/run_benchmark.sh

```diff
@@ -10,115 +10,273 @@
+    -o $rapids_jar
+fi
+
+
+spark_rapids_confs=$( 
+cat <<EOF 
+--spark_confs spark.executorEnv.PYTHONPATH=${rapids_jar} \
+--spark_confs spark.sql.files.minPartitionNum=${num_gpus} \
+--spark_confs spark.rapids.memory.gpu.minAllocFraction=0.0001 \
+--spark_confs spark.plugins=com.nvidia.spark.SQLPlugin \
+--spark_confs spark.locality.wait=0s \
+--spark_confs spark.sql.cache.serializer=com.nvidia.spark.ParquetCachedBatchSerializer \
+--spark_confs spark.rapids.memory.gpu.pooling.enabled=false \
+--spark_confs spark.rapids.sql.explain=ALL \
+--spark_confs spark.rapids.memory.gpu.reserve=20 \
+--spark_confs spark.sql.execution.sortBeforeRepartition=false \
+--spark_confs spark.rapids.sql.format.parquet.reader.type=MULTITHREADED \
+--spark_confs spark.rapids.sql.format.parquet.multiThreadedRead.maxNumFilesParallel=20 \
+--spark_confs spark.rapids.sql.multiThreadedRead.numThreads=20 \
+--spark_confs spark.rapids.sql.python.gpu.enabled=true \
+--spark_confs spark.rapids.memory.pinnedPool.size=2G \
+--spark_confs spark.python.daemon.module=rapids.daemon \
+--spark_confs spark.rapids.sql.batchSizeBytes=512m \
+--spark_confs spark.sql.adaptive.enabled=false \
+--spark_confs spark.sql.files.maxPartitionBytes=2000000000000 \
+--spark_confs spark.rapids.sql.concurrentGpuTasks=2 \
+--spark_confs spark.jars=${rapids_jar}
+EOF
+)
+fi
+    
 # KMeans
-if [[ "${MODE}" == "kmeans" ]] || [[ "${MODE}" == "all" ]]; then
-    if [[ ! -d "/tmp/blobs/5k_3k_float32.parquet" ]]; then
-        python ./benchmark/gen_data.py blobs \
-            --num_rows 5000 \
-            --num_cols 3000 \
+if [[ "${MODE}" =~ "kmeans" ]] || [[ "${MODE}" == "all" ]]; then
+    if [[ ! -d "${gen_data_root}/default/r${num_rows}_c${num_cols}_float32.parquet" ]]; then
+        python $gen_data_script default \
+            --num_rows $num_rows \
+            --num_cols $num_cols \
+            --output_num_files $output_num_files \
+            --numPartitions $output_num_files \
             --dtype "float32" \
             --feature_type "array" \
-            --output_dir "/tmp/blobs/5k_3k_float32.parquet" \
-            --spark_conf "spark.master=local[4]" \
-            --spark_confs "spark.driver.memory=128g"
+            --output_dir "${gen_data_root}/default/r${num_rows}_c${num_cols}_float32.parquet" \
+            $common_confs
+           
     fi
 
+    echo "$sep algo: kmeans $sep"
     python ./benchmark/benchmark_runner.py kmeans \
-        --k 3 \
-        --num_gpus 1 \
-        --num_cpus 0 \
-        --train_path "/tmp/blobs/5k_3k_float32.parquet" \
-        --report_path "report_kmeans.csv" \
-        --spark_confs "spark.master=local[4]" \
-        --spark_confs "spark.driver.memory=128g" \
-        --spark_confs "spark.sql.execution.arrow.maxRecordsPerBatch=20000" \
+        --k 1000 \
+        --tol 1.0e-20 \
+        --maxIter 30 \
+        --initMode random \
```

Comment:
Any specific reason why these param values were chosen?  i.e. should we document that reason here?  Same with all other algos.

Reply:
The args now match what was used in our blog post and the cloud benchmarks.   High compute regime.